server/neptune/temporalworker/server.go

```diff
@@ -104,39 +104,44 @@ func NewServer(config *Config) (*Server, error) {
 }
 
 func (s Server) Start() error {
-	var wg sync.WaitGroup
-	wg.Add(1)
 	defer s.Logger.Close()
 	defer s.TemporalClient.Close()
 
-	go func() {
-		defer wg.Done()
-		w := worker.New(s.TemporalClient, workflows.DeployTaskQueue, worker.Options{
+	// we create a base context that is marked done when we get a sigterm.
+	// we should use this context for other async work to ensure we
+	// are gracefully handling shutdown and not dropping data.
+	mainCtx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+	defer stop()
+
+	// error group here makes it easier to add other processes and share a ctx between them
+	group, gCtx := errgroup.WithContext(mainCtx)
+	group.Go(func() error {
+		w := worker.New(s.TemporalClient, workflows.DeployTaskQueue, worker.Options{})
+		if err := w.Run(worker.InterruptCh()); err != nil {
+			log.Fatalln("unable to start worker", err)
+		}
+		return nil
+	})
+	group.Go(func() error {
+		w := worker.New(s.TemporalClient, workflows.TerraformTaskQueue, worker.Options{
 			// ensures that sessions are preserved on the same worker
 			EnableSessionWorker: true,
 		})
 		if err := w.Run(worker.InterruptCh()); err != nil {
```

Comment:
Hmm kinda weird that we're still using an interruptchannel here when we're keeping this in a shared error group with a context, but it seems like it would still work fine?

Reply:
Yeah, the only other way to start a worker up was non-blocking: https://pkg.go.dev/go.temporal.io/sdk/worker#Worker. 

Wanted to test it a few times and see shutdown behavior in action. Alternatively will just revert back to regular go routines for this server.

Reply:
I reverted back to go routines. I don't think go groups fit our use case if we can already stop the workers by using the interrupt channel, and I didn't want the go group to exit in the case it handles the context cancel before the internal `w.Run()` call did.